passes/techmap/muxcover.cc

```diff
@@ -446,19 +457,26 @@ struct MuxcoverWorker
 				mux.inputs.push_back(O);
 				mux.inputs.push_back(P);
 
-				mux.cost += prepare_decode_mux(S1, S2, T1, bit);
-				mux.cost += prepare_decode_mux(S3, S4, T2, bit);
-				mux.cost += prepare_decode_mux(S5, S6, T3, bit);
-				mux.cost += prepare_decode_mux(S7, S8, T4, bit);
-				mux.cost += prepare_decode_mux(S1, S3, U1, bit);
-				mux.cost += prepare_decode_mux(S5, S7, U2, bit);
-				mux.cost += prepare_decode_mux(S1, S5, V1, bit);
+				mux.cost += prepare_decode_mux(U1, U2, V1, bit);
 
 				mux.cost += prepare_decode_mux(T1, T2, U1, bit);
 				mux.cost += prepare_decode_mux(T3, T4, U2, bit);
-				mux.cost += prepare_decode_mux(T1, T3, V1, bit);
+				if (U1 != U2 && U1 != State::Sx && U2 != State::Sx) {
+					mux.cost += prepare_decode_mux(T1, T3, V1, bit);
 
-				mux.cost += prepare_decode_mux(U1, U2, V1, bit);
+					mux.cost += prepare_decode_mux(S1, S2, T1, bit);
```

Comment:
TODO: Perhaps this shouldn't be nested inside `if (U1 != U2 && U1 != State::Sx && U2 != State::Sx)`